### PR TITLE
fix(dedicated): wip bandwidth link fix

### PR DESCRIPTION
**Pull request by sachinrameshn**
ref: MANAGER-13955

| Question         | Answer
| ---------------- | ---
| Branch?          | `multi-az-feature` 
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | MANAGER-13955 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [ ] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [x] Breaking change is mentioned in relevant commits

### CODE REVIEW

File: packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/clusters/nodes/node/dashboard/dashboard.routing.js [hunk 1]
```diff
@@ -241,12 +241,15 @@ export default /* @ngInject */ ($stateProvider) => {
             productId: serverName,
           },
         ),
-      orderPrivateBandwidthLink: /* @ngInject */ ($state, serverName) =>
+      getOrderPrivateBandwidthLink: /* @ngInject */ (
+        $state,
+        serverName,
+      ) => () =>
```
Comment (darsene) [inline packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/clusters/nodes/node/dashboard/dashboard.routing.js:247]: Why do you need to create a function ? It should be OK with the code as is (with the property)

File: packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/clusters/nodes/node/dashboard/dashboard.routing.js [hunk 2]
```diff
@@ -241,12 +241,15 @@ export default /* @ngInject */ ($stateProvider) => {
...
         $state.href(
           'app.dedicated-cluster.cluster.node.dashboard.bandwidth-private-order',
           { productId: serverName },
         ),
-      orderPublicBandwidthLink: /* @ngInject */ ($state, serverName) =>
+      getOrderPublicBandwidthLink: /* @ngInject */ ($state, serverName) => () =>
```
Comment (darsene) [inline packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/clusters/nodes/node/dashboard/dashboard.routing.js:252]: see https://github.com/ovh/manager/pull/11172#discussion_r1523590144

File: packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/servers/server/dashboard/dashboard.routing.js [hunk 3]
```diff
@@ -207,11 +207,11 @@ export default /* @ngInject */ ($stateProvider) => {
         $state.go('app.dedicated-server.server.dashboard.monitoringUpdate', {
           productId: serverName,
         }),
-      orderPrivateBandwidthLink: /* @ngInject */ (
+      getOrderPrivateBandwidthLink: /* @ngInject */ (
```
Comment (darsene) [inline packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/servers/server/dashboard/dashboard.routing.js:210]: see https://github.com/ovh/manager/pull/11172#discussion_r1523590144

File: packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/servers/server/dashboard/dashboard.routing.js [hunk 4]
```diff
@@ -221,11 +221,11 @@ export default /* @ngInject */ ($stateProvider) => {
               'app.dedicated-server.server.dashboard.bandwidth-private-order',
               { productId: serverName },
             ),
-      orderPublicBandwidthLink: /* @ngInject */ (
+      getOrderPublicBandwidthLink: /* @ngInject */ (
```
Comment (darsene) [inline packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/servers/server/dashboard/dashboard.routing.js:224]: see https://github.com/ovh/manager/pull/11172#discussion_r1523590144

File: packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/clusters/nodes/node/dashboard/dashboard.routing.js (hunk 1 above)
Comment (sachinrameshn) [reply packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/clusters/nodes/node/dashboard/dashboard.routing.js:247]: Its not able to find the state at initial resolve hence modifying it to be as funtcion.

File: packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/clusters/nodes/bandwidth/public-order/index.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import angular from 'angular';
+
+import '@uirouter/angularjs';
+import 'oclazyload';
+
+const moduleName = 'ovhManagerDedicatedClusterNodeBandwidthLazyLoading';
```
Comment (anooparveti) [inline packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/clusters/nodes/bandwidth/public-order/index.js:6]:
```suggestion
const moduleName = 'ovhManagerDedicatedClusterNodeBandwidthPublicOrderLazyLoading';
```

File: packages/manager/modules/bm-server-components/src/server-dashboard/dashboard.html [hunk 6]
```diff
@@ -137,8 +137,8 @@
                 data-tracking-prefix="$ctrl.trackingPrefix"
                 data-specifications="$ctrl.specifications"
                 data-service-infos="$ctrl.serviceInfos"
-                data-order-private-bandwidth-link="$ctrl.orderPrivateBandwidthLink"
-                data-order-public-bandwidth-link="$ctrl.orderPublicBandwidthLink"
+                data-order-private-bandwidth-link="$ctrl.getOrderPrivateBandwidthLink()"
+                data-order-public-bandwidth-link="$ctrl.getOrderPublicBandwidthLink()"
```
Comment (anooparveti) [inline packages/manager/modules/bm-server-components/src/server-dashboard/dashboard.html:141]:
```suggestion
                data-order-private-bandwidth-link=":: $ctrl.getOrderPrivateBandwidthLink()"
                data-order-public-bandwidth-link=":: $ctrl.getOrderPublicBandwidthLink()"
```

**Review by anooparveti — CHANGES_REQUESTED** [review]

**Review by anooparveti — APPROVED** [review]

**Review by darsene — APPROVED** [review]

**Pull request merged**